src/main/java/marquez/common/models/UrnType.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.common.models;
+
+public enum UrnType {
```

Comment:
Is it necessary to have `Type` in the class name since it is a type?

Reply:
basically, why do we need `UrnType` to enforce what is a valid Urn type (e.g. `dataset`, `datasource`,...)? the concrete classes (e.g. `DatasourceUrn`) already do that

Reply:
Yeah, think you're right, and this might be overkill. If we define a URN class, it assumes a type and therefore defining an `enum` only externalizes how a URN is really created. Also, means it would require more than one file to be touched. I'll go ahead and remove the enum.

Reply:
Agreed